cdc_rsync_server/cdc_rsync_server.cc

```diff
@@ -126,11 +164,78 @@ class PatchTask : public Task {
     status_ = path::SetFileTime(target_filepath_, file_.client_modified_time);
   }
 
+  const std::string base_filepath_;
+  const std::string target_filepath_;
+  const ChangedFileInfo file_;
+  CdcInterface* const cdc_;
+  const bool need_intermediate_file_ = false;
+  const std::string patched_filepath_;
```

Comment:
Declaring those fields `const` will disable the move constructor and move assignment operator. Not sure if that's relevant to you though.

Reply:
Tasks are unique ptrs that shouldn't be copied. Deleting copy constructors and assignment operators just to be sure.